openverse_catalog/dags/common/pr_review_reminders.py

```diff
@@ -0,0 +1,161 @@
+    if "critical" in priority_label:
+        return Urgency("critical", 1)
+    elif "high" in priority_label:
+        return Urgency("high", 2)
+    elif "medium" in priority_label:
+        return Urgency("medium", 4)
+    elif "low" in priority_label:
+        return Urgency("low", 5)
+
+
+def days_without_weekends(today: datetime, delta: timedelta) -> int:
+    days_in_previous_week = abs(today.weekday() - delta.days)
+    if days_in_previous_week > 0:
+        if days_in_previous_week < 2:
+            return 0
+        return abs(delta.days - max((days_in_previous_week // 7) * 2, 2))
+
+    return delta.days
+
+
+def get_urgency_if_urgent(pr) -> Optional[ReviewDelta]:
+    updated_at = datetime.fromisoformat(pr["updated_at"].rstrip("Z"))
+    today = datetime.now()
+    urgency = pr_urgency(pr)
+    days = days_without_weekends(today, today - updated_at)
+
+    return ReviewDelta(urgency, days) if days > urgency.days else None
+
+
+def has_already_reviewed(request, reviews):
+    return request["login"] in [review["user"]["login"] for review in reviews]
+
+
+COMMENT_MARKER = (
+    "This reminder is being automatically generated due to the urgency configuration."
+)
+
+
+COMMENT_TEMPLATE = (
+    """
+Based on the {urgency_label} urgency of this PR, the following reviewers are being
+gently reminded to review this PR:
+
+{user_logins}
+
+"""
+    f"{COMMENT_MARKER}"
+    """
+Ignoring weekend days, this PR was updated {days_since_update} day(s) ago. PRs
+labelled with {urgency_label} urgency are expected to be reviewed within {urgency_days}.
+
+@{pr_author}, if this PR is not ready for a review, please draft it to prevent reviewers
+from getting further unnecessary pings.
+"""
+)
+
+
+def build_comment(review_delta: ReviewDelta, stale_requests, pr):
+    user_handles = [f"@{req['login']}" for req in stale_requests]
+    return user_handles, COMMENT_TEMPLATE.format(
+        urgency_label=review_delta.urgency.label,
+        urgency_days=review_delta.urgency.days,
+        user_logins="\n".join(user_handles),
+        days_since_update=review_delta.days,
+        pr_author=pr["user"]["login"],
+    )
+
+
+def base_repo_name(pr):
+    return pr["base"]["repo"]["name"]
+
+
+def post_reminders(github_pat: str, dry_run: bool):
+    if datetime.now().weekday() >= 5:
+        return  # it's the weekend!
+
+    gh = GitHubAPI(github_pat)
+
+    repositories = [
+        "openverse",
+        "openverse-catalog",
+        "openverse-api",
+        "openverse-frontend",
+        "openverse-infrastructure",
+    ]
+
+    open_prs = []
+    for repo in repositories:
+        open_prs += [pr for pr in gh.get_open_prs(repo) if not pr["draft"]]
+
+    urgent_prs = []
+    for pr in open_prs:
+        review_delta = get_urgency_if_urgent(pr)
+        if review_delta:
+            urgent_prs.append((pr, review_delta))
+
+    to_ping = []
+    for pr, review_delta in urgent_prs:
+        comments = gh.get_issue_comments(base_repo_name(pr), pr["number"])
```

Comment:
`base_repo_name(pr)` is called several times, could it be set to a variable early on and reused?

Reply:
No because we're still iterating over a set of PRs that could come from any one of the five repos. I'll see if there's any places where that can be put into a variable for certain blocks but for the most part we have to call it every time we're working on a new PR object.

Reply:
Ah! That makes sense 👍🏼